src/jit/unwind.cpp

```diff
@@ -118,6 +118,299 @@ void Compiler::unwindGetFuncLocations(FuncInfoDsc*             func,
+         regNum           = REG_NEXT(regNum), regBit <<= 1)
+    {
+        if (regBit > regMask)
+        {
+            break;
+        }
+
+        if (regBit & regMask)
+        {
+            unwindPushCFI(regNum);
+        }
+    }
+}
+
+void Compiler::unwindAllocStackCFI(unsigned size)
+{
+    assert(compGeneratingProlog);
+
+    FuncInfoDsc* func = funCurrentFunc();
+
+    unsigned int cbProlog = unwindGetCurrentOffset(func);
+    noway_assert((BYTE)cbProlog == cbProlog);
+    createCfiCode(func, cbProlog, CFI_ADJUST_CFA_OFFSET, DWARF_REG_ILLEGAL, size);
+}
+
+//------------------------------------------------------------------------
+// Compiler::unwindSetFrameRegCFI: Record a cfi info for a frame register set.
+//
+// Arguments:
+//    reg    - The register being set as the frame register.
+//    offset - The offset from the current stack pointer that the frame pointer will point at.
+//
+void Compiler::unwindSetFrameRegCFI(regNumber reg, unsigned offset)
+{
+    assert(compGeneratingProlog);
+    FuncInfoDsc* func = funCurrentFunc();
+
+    unsigned int cbProlog = unwindGetCurrentOffset(func);
+    noway_assert((BYTE)cbProlog == cbProlog);
+
+    createCfiCode(func, cbProlog, CFI_DEF_CFA_REGISTER, mapRegNumToDwarfReg(reg));
+    if (offset != 0)
+    {
+        // before: cfa = rsp + old_cfa_offset;
+        //         rbp = rsp + offset;
+        // after: cfa should be based on rbp, but points to the old address:
+        //         rsp + old_cfa_offset == rbp + old_cfa_offset + adjust;
+        // adjust = -offset;
+        int adjust = -(int)offset;
+        createCfiCode(func, cbProlog, CFI_ADJUST_CFA_OFFSET, DWARF_REG_ILLEGAL, adjust);
+    }
+}
+
+void Compiler::unwindEmitFuncCFI(FuncInfoDsc* func, void* pHotCode, void* pColdCode)
+{
+    UNATIVE_OFFSET startOffset;
+    UNATIVE_OFFSET endOffset;
+    DWORD          unwindCodeBytes = 0;
+    BYTE*          pUnwindBlock    = nullptr;
+
+    if (func->startLoc == nullptr)
+    {
+        startOffset = 0;
+    }
+    else
+    {
+        startOffset = func->startLoc->CodeOffset(genEmitter);
+    }
+
+    if (func->endLoc == nullptr)
+    {
+        endOffset = info.compNativeCodeSize;
+    }
+    else
+    {
+        endOffset = func->endLoc->CodeOffset(genEmitter);
+    }
+
+    DWORD size = (DWORD)func->cfiCodes->size();
+    if (size > 0)
+    {
+        unwindCodeBytes = size * sizeof(CFI_CODE);
+        pUnwindBlock    = (BYTE*)&(*func->cfiCodes)[0];
+    }
+
+#ifdef DEBUG
+    if (opts.dspUnwind)
+    {
+        DumpCfiInfo(true /*isHotCode*/, startOffset, endOffset, unwindCodeBytes, (const CFI_CODE* const)pUnwindBlock);
+    }
+#endif // DEBUG
+
+    assert(endOffset <= info.compTotalHotCodeSize);
+
+    eeAllocUnwindInfo((BYTE*)pHotCode, nullptr /* pColdCode */, startOffset, endOffset, unwindCodeBytes, pUnwindBlock,
+                      (CorJitFuncKind)func->funKind);
+
+    if (pColdCode != NULL)
```

Comment:
This NULL needs to be changed to nullptr for clang-tidy to pass

Reply:
Added PR https://github.com/dotnet/coreclr/pull/14602

Reply:
@benaadams Thanks!